generator_test.go

```diff
@@ -376,6 +378,103 @@ func testNewV5DifferentNamespaces(t *testing.T) {
 	}
 }
 
+func Test_defaultHWAddrFunc(t *testing.T) {
+	tests := []struct {
+		n  string
+		fn func() ([]net.Interface, error)
+		hw net.HardwareAddr
+		e  string
+	}{
+		{
+			n: "error",
+			fn: func() ([]net.Interface, error) {
+				return nil, errors.New("controlled failure")
+			},
+			e: "controlled failure",
+		},
+		{
+			n: "no_valid_hwaddr_returned",
```

Comment:
Other subtests in this package are named in `PascalCase`, such as `FaultyRand`, `StaleEpoch`, etc. Let's keep this consistent and not use `snake_case`.